server/server.go

```diff
@@ -56,14 +58,30 @@ func Start(cfg *Config) error {
 		return errors.Wrap(err, "Failed to setup HTTP client")
 	}
 
-	samlSP, err := samlsp.New(samlsp.Options{
-		URL:            *rootUrl,
-		Key:            keyPair.PrivateKey.(*rsa.PrivateKey),
-		Certificate:    keyPair.Leaf,
-		IDPMetadataURL: idpMetadataUrl,
-		CookieDomain:   rootUrl.Hostname(),
-		HTTPClient:     httpClient,
-	})
+	samlOpts := samlsp.Options{
+		URL:          *rootUrl,
+		Key:          keyPair.PrivateKey.(*rsa.PrivateKey),
+		Certificate:  keyPair.Leaf,
+		CookieDomain: rootUrl.Hostname(),
+		HTTPClient:   httpClient,
+	}
+
+	if idpMetadataUrl.Scheme == "file" {
+		data, err := ioutil.ReadFile(idpMetadataUrl.Path)
+		if err != nil {
+			return errors.Wrap(err, "Failed to read IdP metadata file.")
```

Comment:
One of these days I should switch all these out with the new fmt.Errorf %w approach